lib/Mojo/Asset/File.pm

```diff
@@ -128,6 +133,13 @@ sub slurp {
 
 sub to_file {shift}
 
+sub _end_range {
+  my $file = shift;
+  return undef unless $file->is_range;
+  return $file->end_range if $file->end_range;
```

Comment:
Calling the same method twice is very inefficient.

Reply:
Now directly accessing hash element.